bob/src/hw_metrics_collector.rs

```diff
@@ -47,16 +47,17 @@ impl HWMetricsCollector {
 
     async fn task(t: Duration, disks: HashMap<PathBuf, String>) {
         let mut interval = interval(t);
-        let mut sys = System::new_all();
+        let total_mem = {
+            let sys = System::new_all();
+            kb_to_mb(sys.total_memory())
+        };
         let mut dcounter = DescrCounter::new();
-        let total_mem = kb_to_mb(sys.total_memory());
         debug!("total mem in mb: {}", total_mem);
         let pid = std::process::id() as i32;
 
         loop {
             interval.tick().await;
-            sys.refresh_all();
-            sys.refresh_disks();
+            let sys = System::new_all();
```

Comment:
I did measure it, and it is less than 400ms per call. With the cycle of 60secs it's negligible (less than 1% of the cycle). But if we are talking about more frequent calls than yes, it may be better to keep that structure permanently open.

Reply:
400 ms is a quite long time. The procedure can use CPU or disk resources during initialization, so it can have a noticeable impact on the performance of Bob `Get` and `Put`. I think, that 1000 open descriptors is the lesser evil, so I suggest to leave everything as it was before (a single structure for the lifetime of the application)